src/dune_rules/coq/coq_rules.ml

```diff
@@ -307,19 +268,30 @@ let boot_type ~dir ~use_stdlib ~wrapper_name coq_module =
       if init then `Bootstrap_prelude else `Bootstrap lib
   else `Bootstrap_prelude
 
-let setup_coqdep_rule ~sctx ~dir ~loc ~theories_deps ~wrapper_name ~use_stdlib
-    ~source_rule ~ml_flags ~mlpack_rule coq_module =
-  let* boot_type = boot_type ~dir ~use_stdlib ~wrapper_name coq_module in
+let dep_theory_file ~dir ~wrapper_name =
+  Path.Build.relative dir wrapper_name
+  |> Path.Build.set_extension ~ext:".theory.d"
+
+let setup_coqdep_for_theory_rule ~sctx ~dir ~loc ~theories_deps ~wrapper_name
+    ~use_stdlib ~source_rule ~ml_flags ~mlpack_rule coq_modules =
+  let* boot_type =
+    (* TODO find the boot type a better way *)
+    boot_type ~dir ~use_stdlib ~wrapper_name (List.hd coq_modules)
```

Comment:
This indeed requires a fixme, to start with it will segfault if doing `(modules )` , that is to say, a set of empty modules.

The right way IMHO to do this is to re-introduce `Bootstrap.t` and separate flags from dep injection.